### PR TITLE
fix: remove read messages after migration [AR-3237]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3237" title="AR-3237" target="_blank">BugAR-3237</a>  Conversations are being unread after migration when they were read before
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

After moving unread events to different table, after migration all messages are marked as unread.

### Solutions

Because flow now works different, we need to trigger marking messages as read after messages migration 

##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by vitorhugods — COMMENTED** [review]
Could you please fill the PR description before requesting reviews? 😅

File: app/src/main/kotlin/com/wire/android/migration/userDatabase/ScalaMessageDAO.kt [hunk 1]
```diff
@@ -101,13 +102,13 @@ class ScalaMessageDAO(
...
 
-    private fun messagesFromConversation(scalaConversation: ScalaConversationData): List<ScalaMessageData> {
+    private fun messagesFromConversation(scalaConversation: Conversation): List<ScalaMessageData> {
         val cursor = db.rawQuery(
             "SELECT *" + // Assets are required, otherwise we get exception "requesting column name with table name".
                     "FROM $MESSAGES_TABLE_NAME AS m " +
                     "LEFT JOIN $ASSETS_TABLE_NAME ON m.$COLUMN_ASSET_ID = $ASSETS_TABLE_NAME.$COLUMN_ID " +
                     "WHERE m.$COLUMN_CONVERSATION_ID = ?" +
-                    "ORDER BY m.$COLUMN_TIME ASC ", arrayOf(scalaConversation.id)
+                    "ORDER BY m.$COLUMN_TIME ASC ", arrayOf(scalaConversation.id.value)
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/migration/userDatabase/ScalaMessageDAO.kt]: Holly.

Was this working before with `scalaConversation.id`? 🤔

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1581/1/) **failed**.

**Comment by Garzas** [comment]
> Could you please fill the PR description before requesting reviews? 😅

That's why I didn't requested reviews @vitorhugods 😄

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1581/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6094273)(6094273)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6094273)(6094273)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1581/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6094495)(6094495)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6094495)(6094495)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt [hunk 2]
```diff
@@ -126,19 +130,28 @@ class MigrationManager @Inject constructor(
...
                         Either.Left(failedConversation.values.first())
                     }
                 }
-            }.fold({
-                when (it) {
-                    is NetworkFailure.NoNetworkConnection -> MigrationData.Result.Failure.NoNetwork
-                    else -> MigrationData.Result.Failure.Messages(it.getErrorCode().toString())
+            }
+                .map {
+                    appLogger.d("$TAG - Step 6 - Clean read messages for ${userId.value.obfuscateId()}")
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt:143]: This step should also be applied in `migrateAccountsData`,  this `migrateSingleUser` method is used for a single user after the install, but there is another method `migrate` which is called right after the update and uses `migrateAccountsData` 😉 
Maybe we could extract the common code 🤔

File: app/src/main/kotlin/com/wire/android/migration/userDatabase/ScalaMessageDAO.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/migration/userDatabase/ScalaMessageDAO.kt:111]: Previously it was `ScalaConversationData` and now it uses `Conversation`, that's why it worked before 😄

File: app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt (hunk 2 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt:143]: Good to know 👁️

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1581/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6112100)(6112100)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6112100)(6112100)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt [hunk 3]
```diff
@@ -161,6 +158,17 @@ class MigrationManager @Inject constructor(
             globalDataStore.setUserMigrationStatus(userId.value, UserMigrationStatus.Completed)
         }
 
+    private suspend fun clearReadMessages(
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt]: Shouldn't it be called "clear unread messages"? 🤔 
I would just add a comment describing why it's needed for people who aren't familiar with changes from this PR. 😄

**Review by saleniuk — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/migration/userDatabase/ScalaMessageDAO.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/migration/userDatabase/ScalaMessageDAO.kt]: Previously `ScalaConversationData` was used and it was changed to `Conversation`, that's why it worked before 😄

**Review by yamilmedina — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt (hunk 3 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt]: Good point

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1581/7/) **failed**.

**Review by MohamadJaara — APPROVED** [review]
LGTM

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1581/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.0(6129170)(6129170)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.0(6129170)(6129170)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1581/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.0(6129475)(6129475)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.0(6129475)(6129475)-staging-compat.apk)

**Pull request merged**